tests/positive/partitions/zeros.go

```diff
@@ -0,0 +1,246 @@
+// Copyright 2018 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package partitions
+
+import (
+	"github.com/coreos/ignition/tests/register"
+	"github.com/coreos/ignition/tests/types"
+)
+
+func init() {
+	// Tests that just create partitions and involve 0s
+	register.Register(register.PositiveTest, PartitionSizeStart0())
+	register.Register(register.PositiveTest, PartitionStartNumber0())
+	register.Register(register.PositiveTest, ResizeRootFillDisk())
+	register.Register(register.PositiveTest, VerifyRootFillsDisk())
+	register.Register(register.PositiveTest, VerifyUnspecifiedIsDoNotCare())
+}
+
+func PartitionSizeStart0() types.Test {
+	name := "Create a partition with size and start 0"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	config := `{
+		"ignition": {
+			"version": "2.1.0"
+		},
+		"storage": {
+			"disks": [{
+				"device": "$disk1",
+				"wipeTable": false,
+				"partitions": [{
+					"label": "fills-disk",
+					"number": 1,
+					"start": 0,
+					"size": 0,
+					"typeGuid": "F39C522B-9966-4429-A8F8-417CD5D83E5E",
+					"guid": "3ED3993F-0016-422B-B134-09FCBA6F66EF"
+				}]
+			}]
+		}
+	}`
+
+	in = append(in, types.Disk{
+		Alignment: types.IgnitionAlignment,
+	})
+	out = append(out, types.Disk{
+		Alignment: types.IgnitionAlignment,
+		Partitions: types.Partitions{
+			{
+				Label:    "fills-disk",
+				Number:   1,
+				Length:   65536,
+				TypeGUID: "F39C522B-9966-4429-A8F8-417CD5D83E5E",
+				GUID:     "3ED3993F-0016-422B-B134-09FCBA6F66EF",
+			},
+		},
+	})
+
+	return types.Test{
+		Name:   name,
+		In:     in,
+		Out:    out,
+		Config: config,
+	}
+}
+
+func PartitionStartNumber0() types.Test {
+	name := "Create a partition with size and start 0"
+	in := types.GetBaseDisk()
+	out := types.GetBaseDisk()
+	config := `{
+		"ignition": {
+			"version": "2.1.0"
```

Comment:
Is a `2.1.0` config intentional?

Reply:
In this case, yes, since the config uses only old keys.